src/main/java/io/dataspaceconnector/common/util/UUIDUtils.java

```diff
@@ -138,7 +138,7 @@ public static UUID createUUID(final Function<UUID, Boolean> doesUuidExistFunc,
             final var uuid = UUID.randomUUID();
 
             // Check if the created uuid already exists
-            if (!doesUuidExistFunc.apply(uuid)) {
+            if (Boolean.FALSE.equals(doesUuidExistFunc.apply(uuid))) {
```

Comment:
Why is this an advantage over the old code?

Reply:
`.apply` implies that it is functional programming. Thus, in theory, we do not know what `doesUuidExistFunc` returns, which could also be `null`. This would lead to `if (!null)`.

Reply:
Very broken down, this goes back to the difference between primitive `boolean` and `Boolean`. The following states are valid:
`boolean: true/false`
`Boolean: true/false/null`
Therefore, a function that returns a `Boolean` can also return `null`.